src/GoogleAdWordsEventForwarder.js

```diff
@@ -269,6 +396,31 @@
             return true;
         }
 
+        // Creates a new Consent State Payload based on Consent State and Mapping
+        function generateConsentStatePayloadFromMappings(consentState, mappings, defaults) {
+            const payload = cloneObject(defaults);
+    
+            for (var mappingEntry of mappings) {
+                if (
+                    consentState[mappingEntry.map] &&
+                    mappingEntry.maptype === 'ConsentPurposes' &&
+                    googleConsentProperties.includes(mappingEntry.value)
+                ) {
+                    payload[mappingEntry.value] = consentState[
+                        mappingEntry.map
+                    ].Consented
+                        ? googleConsentValues.Granted
+                        : googleConsentValues.Denied;
+                }
+            }
+    
+            return payload;
+        }
+
+        function stringifyPayload(payload) {
+            return Object.entries(payload).join(',');
```

Comment:
Object.entries is es6.  I think you'll have to alphabetize Object.keys for a predictable order, and loop over that.  You can also try to place it in an array based on the order of `googleConsentProperties` and join that.